Add CI for support-ar6.0.z

akariiijima opened the pull request:
### Goodbye TravisCI, Welcome GithubActions
TravisCI is currently not working. This is probably because it was not maintained. Therefore, we will adopt GithubActions instead of using TravisCI.

### CI for support-ar6.0.z
Matrix build test for ruby 2.4-2.7 ✖️ rails 5.0.0-6.0.6

### Fix gemfiles for some reasons about dependencies
- Add gem rails for `require "rails"` in spec/spec_helper.rb
- Use gem mysql2 0.4.x less than rails 5.1.6, and use gem mysql2 0.5.x more than rails 5.1.6
  - https://github.com/rails/rails/blob/5-1-stable/activerecord/CHANGELOG.md#rails-516-march-29-2018
- In ruby 2.4 or less, there are several gem dependencies
  - nokogiri
    - `bundle install` will install a native gem without any additional action in more then ruby 2.5, but in ruby 2.4 or less, need to add gem nokogiri.
      - https://github.com/sparklemotion/nokogiri
    - With the addition of the above adding gem nokogiri 1.10, loofah should be pin version 2.20.2
      - https://github.com/flavorjones/loofah/issues/266
      - https://github.com/getsentry/sentry-ruby/pull/2041
  - dalli
    - Add gem dalli for `require "dalli"` in spec/active_record/turntable/sequencer/katsubushi_spec.rb 
    - dalli should be pin version 2
    - https://github.com/petergoldstein/dalli/issues/758

### Some jobs failed because... 

- More than rails 6 support only ruby more than 2.5, so ruby 2.4 or less is failed CI

hirocaster commented on gemfiles/rails5_0_0.gemfile:
```diff
@@ -1,10 +1,14 @@
 source "https://rubygems.org"
 
+gem "rails", "5.0.0"
```
Why did you need to add a gem such as Rails?

hirocaster replied:
I asked to be listed in the DESCRIPTION.

akariiijima replied:
https://github.com/mixigroup/activerecord-turntable/pull/18#issuecomment-1610825952

hirocaster commented on gemfiles/rails5_1_4.gemfile:
```diff
@@ -1,12 +1,14 @@
 
-gem "minitest", "< 5.3.4"
-gem "mocha", "~> 0.14", require: false
+gem "nokogiri", "~> 1.10" if RUBY_VERSION.to_f < 2.5
```
Why are these making changes?

hirocaster replied:
I asked to be listed in the DESCRIPTION.

hirocaster replied:
https://github.com/mixigroup/activerecord-turntable/pull/18#issuecomment-1609045294

hirocaster commented:
Glad you sent us a PR, and we will be sure to keep you posted on our progress. 😄 
However, it would be very helpful if you could explain the reasons for the increase in dependence as it seems to be increasing.

akariiijima commented:
@hirocaster 
I added descriptions about fix reasons! 
Please check and let me know if you have some questions.

hirocaster commented:
https://github.com/mixigroup/activerecord-turntable/pull/18#discussion_r1239323534

> Why did you need to add a gem such as Rails?

I looked at the DESCRIPTION but could not find a reason for this.

akariiijima commented:
About adding "gem rails", I referenced original file (ex: gemfiles/rails6_0_4.gemfile, etc...)😄

hirocaster commented:
Okay, I understand.
I recognized that I may need to organize my gemfile as it should not inherently depend on Rails.

akariiijima committed:
Add rails 6.0.1, 6.0.2, 6.0.3, 6.0.5, 6.0.6 gemfiles

akariiijima committed:
Pin gem nokogiri/loofah version if ruby < 2.5

bundle install will install a native gem without any additional action : https://github.com/sparklemotion/nokogiri

loofah should be pin version 2.20.2 :
https://github.com/flavorjones/loofah/issues/266, https://github.com/getsentry/sentry-ruby/pull/2041

akariiijima committed:
Add gem dalli because dalli is needed in spec

Pin gem dalli version 2 if ruby < 2.5 : https://github.com/petergoldstein/dalli/issues/758

akariiijima committed:
Add workflows for ruby 2.4 - 2.7 in GithubActions

akariiijima commented:
`gem rails` is needed for `require rails`
https://github.com/mixigroup/activerecord-turntable/blob/3d4a955b9ef7e15765a0ef0ec4670413939a1573/spec/spec_helper.rb#L6
Gemfiles seem to be need not to change in this last commit. :+1:

akariiijima committed:
Delete gem minitest because minitest is a testing library built into rails by default

akariiijima committed:
Delete gem mocha because of not being used now

akariiijima committed:
Add gem mysql ~> 0.4.4 to rails 5.0.7

akariiijima committed:
Add gem mysql ~> 0.5.2 to rails 5.1.6 (mysql 0.5.X is supported for more than rails 5.1.6)

akariiijima committed:
Add rails 5.1.7, 5.2.2, 5.2.3, 5.2.4, 5.2.5, 5.2.6, 5.2.7, 5.2.8 gemfiles

akariiijima committed:
Add gem rails because rails is needed in spec

railties and actionview gems have been added to gemfiles not having them. This is because they are dependencies of rails.

akariiijima committed:
Goodbye travisCI because of not being maintained, and use GithubActions instead

hirocaster commented on spec/active_record/turntable/active_record_ext/query_cache_spec.rb:
```diff
@@ -6,16 +6,18 @@ def middleware(&app)
-    ensure
-      if ActiveRecord::Turntable::Util.ar60_or_later?
-        ActiveRecord::Base.connection_handlers = original_handlers
+      begin
```
This commit is duplicate #19.
Please, fix the change commits.

akariiijima replied:
I fixed!
